plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/CameraUpdateProvider.java

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdate;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+class CameraUpdateProvider implements RefreshPluginRunnable.RefreshPluginListener {
```

Comment:
Generally, what about implementing the `RefreshPluginRunnable.RefreshPluginListener` in the `LocationLayerCamera`? That would simplify the checks needed for the animation and reflect the setup that we have with the `LocationLayer`.